lib/Service/ModelService.php

```diff
@@ -48,16 +49,33 @@ public function __construct(IConfig     $config,
 		$this->config     = $config;
 		$this->appData    = $appData;
 		$this->rootFolder = $rootFolder;
-
-		// Construct root folder for models
-		$this->prepareAppDataFolders();
+		
+		// Set model path.
+		setModelPath();
 
 		/// Get this folder.
 		$instanceId = $this->config->getSystemValue('instanceid', null);
 		$appData = $this->rootFolder->get('appdata_'.$instanceId)->getPath();
 		$dataDir = $this->config->getSystemValue('datadirectory', null);
+	}
 
-		$this->modelsFolder = $dataDir . $appData . '/facerecognition/models/';
+	/**
+	 * @return void
+	 */
+	private function setModelPath() {
```

Comment:
Rename `getModelPath()` that return string, and use on constructor.

```
private function getModelPath(): string {
    if (!is_null($modelPath)) {
       return $modelPath;
    }

    $instanceId = $this->config->getSystemValue('instanceid', null);
    $appData = $this->rootFolder->get('appdata_'.$instanceId)->getPath();
    $dataDir = $this->config->getSystemValue('datadirectory', null);

    return $dataDir . $appData . '/facerecognition/models/';
}
```

Reply:
What is the prepareAppdataFolders function for?
it seems to create a 'model' directory on the appdata_{id}/models

Reply:
> What is the prepareAppdataFolders function for?
> It seems to create a 'model' directory under appdata_{id}.
> However, shouldn't it be under appdata_{id}/facerecognition?

Yes.. it's just to create the folder 'facerecognition/models' within appData.

Reply:
```
private function getModelPath(): string {
    if (!is_null($modelPath)) {
       return $modelPath;
    }

    try {
        $this->appData->getFolder('/models');
    } catch (NotFoundException $e) {
         $this->appData->newFolder('/models');
    }

    $instanceId = $this->config->getSystemValue('instanceid', null);
    $appData = $this->rootFolder->get('appdata_'.$instanceId)->getPath();
    $dataDir = $this->config->getSystemValue('datadirectory', null);

    return $dataDir . $appData . '/facerecognition/models/';
}
```

Reply:
> Yes.. it's just to create the folder 'facerecognition/models' within appData.

I believe that needs to be addressed then?

Reply:
Done!
Recursive and permissions 0770 added.
The default 0777 seems unnecessarily open.